src/assets/src/components/preferences.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from "react";
+import { useState, useEffect } from "react";
+import PhoneInput from "react-phone-input-2";
+import 'react-phone-input-2/lib/style.css'
+
+import * as api from "../services/api";
+import { User } from "../models";
+import { ErrorDisplay, LoadingDisplay, LoginDialog, Breadcrumbs } from "./common";
+import { usePromise } from "../hooks/usePromise";
+import { redirectToLogin } from "../utils";
+import { PageProps } from "./page";
+
+interface PreferencesEditorProps {
+    user: User;
+    disabled: boolean;
+    onUpdateInfo: (phoneNumber: string) => void;
+}
+
+function PreferencesEditor(props: PreferencesEditorProps) {
+    var currentPhoneField = ""
```

Comment:
It looks like you're trying to persist state. Did this come from example code somewhere? It seems to be working when I run it, but I think a bug would manifest pretty quickly because of the (somewhat weird) way React handles state. If you look up the `useState` hook and see the way I'm using it in other components, that's the template that I think you should be following in this spot.

Reply:
This was the first way I thought of keeping track of what the user has typed in the phone number field without having to update the user's phone number with every key stroke. I will definitely look at how you've used the `useState` hook and adapt this.